pkgs/development/compilers/gcc/common/configure-flags.nix

```diff
@@ -171,9 +173,8 @@ let
       then ["--enable-multilib" "--disable-libquadmath"]
       else ["--disable-multilib"])
     ++ lib.optional (!enableShared) "--disable-shared"
-    ++ [
-      (lib.enableFeature enablePlugin "plugin")
-    ]
+    ++ [ (lib.enableFeature enablePlugin "plugin") ]
+    ++ [ (lib.enableFeature enableGdbPlugin "libcc1") ]
```

Comment:
Looking at https://gcc.gnu.org/git/?p=gcc.git;a=blob;f=libcc1/configure.ac;h=e9f6a13e17210a2965be959c6c44f896cde19c59;hb=HEAD looks like `libcc1` does not work for static static-only builds (should be the same limitation as plugins) and for targets that don't have UNIX socket APIs (also sound niche).

Generally I would suggest only providing `--disable-<foo>` variant and never use `--enable-<variant>` to rely on defaults for more obscure libraries as they frequently have somewhat complex support matrix.

Should be OK for this case. The only minor annoyance is the rebuilds that probably could be avoided by  `--disable-<foo>`.

Reply:
> looks like `libcc1` does not work for static static-only builds (should be the same limitation as plugins)

Yes; there is an `assert enableGdbPlugin -> enablePlugin`.  Are you suggesting some kind of `assert enablePlugin -> enableShared` as well?

> and for targets that don't have UNIX socket APIs (also sound niche).

I wonder if there are any platforms that support GDB but don't have UNIX sockets!  Maybe mingw?

Reply:
> Generally I would suggest only providing `--disable-<foo>` variant and never use `--enable-<variant>` to rely on defaults for more obscure libraries as they frequently have somewhat complex support matrix.

Oh I get it; you see `enableFoo` as meaning not "enable `Foo`", but rather as "pass the `--enable-foo` flag, otherwise don't pass any `-foo` flag at all".

Interesting.

I guess Nix needs a ternary type (true/false/none) to match `--enable-foo`/`--disable-foo`/`""`.

I will update the PR to make it `disableGdbPlugin`.

Reply:
Ok I switched to `disableGdbPlugin`.

Reply:
Yeah, your change is exactly what I meant. Thank you!